### DISCUSSION TITLE
Allow/Deny popups OK, main UI display never shows in Linux Mint 21-2 (grpcio incompatibility)

### DISCUSSION THREAD

**Discussion by Ekneee** [evilsocket/opensnitch#1003] [Q&A] [answered]
Allow/deny UI popups appear A-OK.  Main GUI 1.6.1 does not and has never appeared.  No error message in terminal window running opensnitch-ui as my user.  

Installed OpenSnitch 1.6.1 in Mint 21-2 Cinnamon after clean Mint 21-2 install from USB thumb drive.

Daemon running OK, per System Monitor with ~0% CPU.  CPU for UI is near 0%.

Tried grpcio 1.56.2 and grpcio 1.44.0 - same result - no main UI.    Followed Issue #647 guidance.
protobuf: 3.8.0  OpenSnitch GUI - 1.6.1

Tried install of both python3-opensnitch-ui_1.6.1-1_all.deb and all.PIP.deb.  Have installed python3-pyasn, no change.  Tried after clean install of Mint 21-2 without and with updating packages before installing OpenSnitch.

Have previously installed OpenSnitch 1.5.1 in Mint 20-3.  Works GREAT.

**Comment by gustavo-iniguez-goya** [comment, marked as answer]
hi @Ekneee ,

I just tested it on Mint 21.2 (latest iso, not updated), and the steps to make it work have been:
 - install opensnitch + opensnitch-ui
 - install pytho3-pip
 - as your regular user install grpcio==1.44.0: `~ $ pip install grpcio==1.44.0` (even without --ignore-installed)

If you don't need `python3-grpcio` for other applications, uninstall it and try again. And if you installed grpcio 1.44.0 with pip system-wide as root, uninstall it and install it as your regular user (with --user).

**Reply by Ekneee** [reply]
OpenSnitch 1.6.1 works A-OK on Mint 21-2 following your instructions.  Many Thanks for the rapid and precisely right response.
